conda_smithy/configure_feedstock.py

```diff
@@ -615,17 +614,36 @@ def _render_ci_provider(
             forge_config,
         )
 
-        metas = conda_build.api.render(
-            os.path.join(forge_dir, forge_config["recipe_dir"]),
-            platform=platform,
-            arch=arch,
-            ignore_system_variants=True,
-            variants=migrated_combined_variant_spec,
-            permit_undefined_jinja=True,
-            finalize=False,
-            bypass_env_check=True,
-            channel_urls=forge_config.get("channels", {}).get("sources", []),
-        )
+        # AFAIK there is no way to get conda build to ignore the CBC yaml
+        # in the recipe. This one can mess up migrators applied with local
+        # CBC yaml files where variants in the migrators are not in the CBC.
+        # Thus we move it out of the way.
+        # TODO: upstream this as a flag in conda-build
```

Comment:
This is just wrong. We need to use the CBC yaml to render the recipe.

Reply:
That is done above here AFAIK: https://github.com/conda-forge/conda-smithy/pull/1454/files#diff-dc66a1980eec62725cfb8b6faea5b3c90dc7d1009d1229f5f55eecfb267a6677R607

Reply:
The CBC will remove variants added by migrators if the keys overlap.

Reply:
See this line in conda build: https://github.com/conda/conda-build/blob/master/conda_build/variants.py#L213

Reply:
I've printed out all of the config variants in each step and without this block, we definitely cannot support rendering migrators with extra keys on top of local CBCs. That is a fine conclusion, but it is the CBC that is removing extra variants for cudnn.

Reply:
Makes sense.